src/wazuh_modules/vulnerability_scanner/src/vulnerabilityReport/vulnerabilityReport.hpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Wazuh vulnerability scanner - Policy Manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * November 22, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _VULNERABILITY_REPORT_HPP
+#define _VULNERABILITY_REPORT_HPP
+
+#include "json.hpp"
+#include "socketClient.hpp"
+#include "vulnerabilityScanner.hpp"
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <sys/epoll.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+// This can be modified according to requirements.
+const std::string LOCALFILE_MQ {"1"};
+const std::string PATH {"/var/ossec/queue/sockets/queue"};
+
+/**
+ * @brief Class in charge to send vulnerability reports.
+ *
+ */
+class VulnerabilityReport
+{
+public:
+    /**
+     * @brief Sends a formatted report through specified socket.
+     *
+     * @param socketClient Socket FD.
+     * @param report JSON object with vulnerability report.
+     * @param agentID Agent ID.
+     * @param agentIP Agent IP.
+     * @param agentName Agent name.
+     */
+    static void sendReport(SocketClient<Socket<OSPrimitives, NoHeaderProtocol>, EpollWrapper>& socketClient,
+                           const nlohmann::json& report,
+                           const std::string& agentID,
+                           const std::string& agentIP,
+                           const std::string& agentName)
+    {
+        try
+        {
+            const std::string reportStr = report.dump();
```

Comment:
This variable is unnecessary (we avoid the string copy). It could be directly concatenated on the message variable.

Reply:
Changed in favor of stringstream.